File: javascript/problem037.js
var array = new Array(1000000);
array[1] = false;

for(i=2; i<Math.sqrt(1000000); i++){
  if(array[i] == undefined){
    for(j=i+i; j<1000000; j = j + i){
      array[j] = false;
    }
  }
}

function isTruncatable(n){
  var str = n.toString();
  for(a=1; a<str.length; a++){
    var tempStr1 = str.substring(0, a);
    var tempStr2 = str.substring(a, str.length)
    var tempValue1 = parseInt(tempStr1);
    var tempValue2 = parseInt(tempStr2);
    if(array[tempValue1] != undefined || array[tempValue2] != undefined){
      return false;
    }
  }
  return true;
}

var result = 0;
var count = 0;
var index = 10;
while(count < 11){
  if(array[index] == undefined && isTruncatable(index)){
    result = result + index;
    count = count + 1;
  }
  index = index + 1;
}

console.log(result);

File: python/problem037.py
import math

number_array = [True] * 1000000
number_array[1] = False

for i in range(2, int(math.sqrt(1000000))):
    if number_array[i]:
        j = i + i
        while j < 1000000:
            number_array[j] = False
            j += i


def is_truncatable(n):
    str_n = str(n)
    for a in range(1, len(str_n)):
        temp_str1 = str_n[0:a]
        temp_str2 = str_n[a:len(str_n)]
        temp_value1 = int(temp_str1)
        temp_value2 = int(temp_str2)
        if not number_array[temp_value1] or not number_array[temp_value2]:
            return False

    return True


result = 0
count = 0
index = 10
while count < 11:
    if number_array[index] and is_truncatable(index):
        result += index
        count += 1
    index += 1
print(result)
